server/api/categoryRouter.js

```diff
@@ -60,13 +60,18 @@ router.get('/getCategoryItems/:categoryId', async (req, res, next) => {
     const response = {};
 
     //get category name and a list of item ids
-    const categoryData = await Category.findOne({ _id: categoryId }, 'title items').exec();
+    const categoryData = await Category.findById(categoryId, 'title items').exec();
+    
+    if (!categoryData) {
+      throw new Error('CategoryId does not exist in database');
+    }
     response.title = categoryData.title;
     const itemIds = categoryData.items;
 
     const itemObjects = await Item.find({ _id: { $in: itemIds } }).exec();
 
     response.items = itemObjects;
+
```

Comment:
nit: perhaps whitespace could be reduced here? I'm not sure it makes the code clearer